context/__tests__/GlobalStore.test.tsx

```diff
@@ -1,7 +1,57 @@
+import { render } from '@testing-library/react';
 import { GlobalStore } from '../GlobalStore';
+import { ActionTypes } from '../reducer';
+import { act } from 'react-test-renderer';
 
-const MockComponent = () => <GlobalStore></GlobalStore>;
+const dispatch = jest.fn();
+
+jest.mock('react', () => {
+  const original = jest.requireActual('react'); // Step 2.
+  return {
+    ...original,
+    useReducer: () => [{}, dispatch],
+  };
+});
 
 describe('GlobalStore', () => {
-  it('', () => {});
+  let originalFetch: any;
+  beforeEach(() => {
+    originalFetch = global.fetch;
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () =>
+          Promise.resolve({
+            searchFilter: { brands: [], categories: [] },
+          }),
+      })
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+    global.fetch = originalFetch;
+  });
+
+  it('fetches data initially', async () => {
+    // render(<GlobalStore />);
+    await act(async () => render(<GlobalStore />));
```

Comment:
Is the act wrapper needed? do you get any warnings without it?

Reply:
Yes, there was warning that we are modifying state in global store without an act, thats why i needed it